### ISSUE TITLE
`font` paramater for `nvim_set_hl()`

### ISSUE THREAD

**Issue by 4513ECHO** [neovim/neovim#27263]
### Problem

The `font` argument for `:highlight` command (`:help highlight-font`) is not supported in `nvim_set_hl()`.

### Expected behavior

`nvim_set_hl()` should support `font` paramater. GUI fontend such as neovide should support the paramater if notified.

**Comment by 4513ECHO** [comment]
usecase: [Monaspace](https://monaspace.githubnext.com/) with Neovim

**Comment by BlueDrink9** [comment]
This would also be a big boon (almost a necessity imo) in a world with variable-width fonts.
*  per-highlight font specifications would solve issues like statuslines (just set it as mono)
* markdown headings could be "highlighted" with a larger font
* markdown that was rendered in the default font (possibly a proportional font) could be forced to monospace for `code blocks`

I think there could be an issue with how to lay out the font specification. As far as I can tell, GUI implementations have a fair bit of free reign in the format of the `guifont` setting. For example, neovide extends it to include height. This current issue may therefore benefit from neovim defining a format for font specifications, including things like height and width, for GUIs and highlight schemes to follow.

**Comment by clason** [comment]
You are putting the cart before the horse. Neovim is first and foremost a TUI application (that is the _only_ client we ship), so all of this is impossible in core. External GUIs intentionally have a lot of leeway, and if a standard emerges, _they_ will tell _us_.

**Comment by BlueDrink9** [comment]
I agree, and there are instances where GUI devs have pushed neovim core in a more GUI friendly direction. My suggestion regarding standards was intended to mean more like "there are many ways GUIs do it, we should pick one and say 'please follow this for highlights', and possibly slowly establish that as a convention for GUIs to use for guifont as well". Without a standard, it becomes impossible for plugin or theme developers to be consistent in how they specify fonts in a highlight group.

**Comment by clason** [comment]
But this is not how the Neovim project works. _Are_ there any GUIs that even allow for local fonts, no matter how they are specified? Standards are driven by implementation, not the other way around.

**Comment by BlueDrink9** [comment]
Well, they couldn't unless they manually arbitrarily assigned them. This would be borderline an api prerequisite, as far as I can see.

Anyway, I was chipping in something I thought might be relevant as this progresses in the future.
